cocos/core/data/object.ts

```diff
@@ -327,7 +327,7 @@ class CCObject {
             this._onPreDestroy();
         }
 
-        if ((TEST ? (/* make EDITOR mockable*/ Function('return !EDITOR'))() : !EDITOR) || cc.engine._isPlaying) {
+        if (!EDITOR || cc.engine._isPlaying) {
```

Comment:
这样会导致单元测试中的 EDITOR 变量无法动态修改，也就无法在单元测试中切换不同执行路径了

Reply:
jest mock?

Reply:
为什么需要动态修改？具体使用的案例在哪里有？

Reply:
![image](https://user-images.githubusercontent.com/1503156/83990549-a6efe580-a97c-11ea-8fab-b4658ac66f7b.png)
Here will test the behavior in the editor

Reply:
I don't think it could be modified at runtime now, we have moved all macros to an internal module, so `_deferredDestroy` is probably using the internal macro EDITOR, not the global one. @shrinktofit is there anyway to simulation such environment dynamically in the unit test ?

Reply:
Fine. Then we should export two version's api for test only.
By calling differenct API to test differenct behaviour.